app/models/subnet.rb

```diff
@@ -207,18 +214,67 @@ def template_proxy(attrs = {})
     @template_proxy ||= ProxyAPI::Template.new({:url => template.url}.merge(attrs)) if template?
   end
 
+  def external_ipam?
+    supports_ipam_mode?(:external_ipam) && externalipam && externalipam.url.present?
+  end
+
+  def external_ipam_proxy(attrs = {})
+    @external_ipam_proxy ||= ProxyAPI::ExternalIpam.new({:url => externalipam.url}.merge(attrs)) if external_ipam?
+  end
+
   def ipam?
     self.ipam != IPAM::MODES[:none]
   end
 
   def ipam_needs_range?
-    ipam? && self.ipam != IPAM::MODES[:eui64]
+    ipam? && self.ipam != IPAM::MODES[:eui64] && self.ipam != IPAM::MODES[:external_ipam]
+  end
+
+  def ipam_needs_group?
+    ipam? && self.ipam == IPAM::MODES[:external_ipam]
   end
 
   def dhcp_boot_mode?
     self.boot_mode == Subnet::BOOT_MODES[:dhcp]
   end
 
+  def validate_against_external_ipam
+    return unless self.errors.full_messages.empty?
+
+    if self.ipam == IPAM::MODES[:external_ipam]
+      external_ipam_proxy = SmartProxy.with_features('External IPAM').first
+
+      if external_ipam_proxy.nil?
+        self.errors.add :ipam, _('There must be at least one Smart Proxy present with an External IPAM plugin installed and configured')
+      elsif self.external_ipam_proxy.nil?
+        self.errors.add :ipam, _('A Smart Proxy with an External IPAM feature enabled must be selected in the Proxies tab.')
+      elsif !group_exists_in_external_ipam
+        self.errors.add :externalipam_group, _('Group not found in the configured External IPAM instance(e.g. "Section" in phpIPAM)')
```

Comment:
Minor nit pick: I'd drop references to imlementations in app itself: "(e.g. "Section" in phpIPAM)".

Reply:
Right - missed that one. I will wait for any review from @ekohl, and update then.